File: Array&string/Product of Array Except Self/ans.py
class Solution:
    def productExceptSelf(self, nums):
        n = len(nums)
        ans = [1] * n

        # Calculating products to the left of each element
        for i in range(1, n):
            ans[i] = ans[i - 1] * nums[i - 1]

        product_right = nums[n - 1]

        # Multiplying by products to the right and updating product_right
        for i in range(n - 2, -1, -1):
            ans[i] *= product_right
            product_right *= nums[i]

        return ans

File: Sliding Window/Minimum Size Subarray Sum/ans.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int ans=INT_MAX;
        int left=0,sum=0;
        int n=nums.size();

        for (int right=0;right<n;right++){
            sum+=nums[right];
            while (sum>=target){
                ans=min(ans,right-left+1);
                sum-=nums[left];
                left++;
            }


        }
        return (ans==INT_MAX)?0:ans;
    }
};